File: Modules/ServerTypes/WorkFlow/RuleColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { RuleRow } from "./RuleRow";

export interface RuleColumns {
    Id: Column<RuleRow>;
    CurrentStatusName: Column<RuleRow>;
    ActionName: Column<RuleRow>;
    NextStatusName: Column<RuleRow>;
    RoleName: Column<RuleRow>;
    IsFinalState: Column<RuleRow>;
    UpdateDate: Column<RuleRow>;
}

export class RuleColumns extends ColumnsBase<RuleRow> {
    static readonly columnsKey = 'WorkFlow.Rule';
    static readonly Fields = fieldsProxy<RuleColumns>();
}

File: Modules/WorkFlow/Rule/RuleColumns.cs
namespace WorkFlow.Columns;

[ColumnsScript("WorkFlow.Rule")]
[BasedOnRow(typeof(RuleRow), CheckNames = true)]
public class RuleColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }    
    public string CurrentStatusName { get; set; }
    public string ActionName { get; set; }
    public string NextStatusName { get; set; }
    public string RoleName { get; set; }
    [Width(100)]
    public bool IsFinalState { get; set; }
    [Width(200)]
    public DateTime UpdateDate { get; set; }
}